smk/gene_annotation.smk

```diff
@@ -362,7 +362,7 @@ rule gene_annot_eggnog:
         time (
             mkdir out
             emapper.py --data_dir {params.eggnog_db} -o tmp \
-                       --no_annot --no_file_comments --report_no_hits --override --output_dir out -m diamond -i {input.faa} --cpu {threads}
+                       --no_annot --no_file_comments --report_no_hits --override --output_dir out -m diamond -i {input.faa} --cpu {threads} {params.eggnog_inmem}
             emapper.py --annotate_hits_table out/tmp.emapper.seed_orthologs \
```

Comment:
The diamond search doesn't have an in-memory option: it's for the annotation sqlite only. (There is an option for HMMER, but that's a different flag.)
https://github.com/eggnogdb/eggnog-mapper/wiki/eggNOG-mapper-v2.1.5-to-v2.1.12#other-requirements